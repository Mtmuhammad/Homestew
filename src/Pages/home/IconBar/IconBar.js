import React from "react";
import "./IconBar.scss";
import { Link } from "react-router-dom";

const SubscribeBar = () => {
  return (
    <div className="icon-bar">
      <div className="container" style={{ maxWidth: "1100px" }}>
        <div className="row">
          <h3 className="tertiary text-center">Explore Recipes:</h3>
          <div className="col-6 col-lg-3">
            <div data-testid="icon-container" className="icon text-center">
              <Link to="recipes/type/bread">
                <i data-testid="icon" className="fa-solid fa-bread-slice"></i>
                <p>Bread Recipes</p>
              </Link>
            </div>
          </div>
          <div className="col-6 col-lg-3">
            <div data-testid="icon-container" className="icon text-center">
              <Link to="recipes/type/main course">
                <i
                  data-testid="icon"
                  className="fa-solid fa-drumstick-bite"
                ></i>
                <p>Main Course</p>
              </Link>
            </div>
          </div>
          <div className="col-6 col-lg-3">
            <div data-testid="icon-container" className="icon text-center">
              <Link to="recipes/type/side dish">
                <i data-testid="icon" className="fa-solid fa-bowl-food"></i>
                <p>Side Dishes</p>
              </Link>
            </div>
          </div>
          <div className="col-6 col-lg-3">
            <div data-testid="icon-container" className="icon text-center">
              <Link to="recipes/type/dessert">
                <i data-testid="icon" className="fa-solid fa-cookie"></i>
                <p>Desserts</p>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SubscribeBar;
