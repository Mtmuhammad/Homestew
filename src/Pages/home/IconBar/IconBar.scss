@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.icon-bar {
  height: auto;
  background-color: $light-primary-green;
  padding: 50px 0;

  .icon {
    margin-top: 3rem;

    & a {
      text-decoration: none;
      color: $primary-font;
      transition: all 0.3s;

      &:hover {
        color: $primary-green;
      }

      & p {
        font-size: 2rem;
        font-weight: 700;
        line-height: 1.1;
        letter-spacing: 0.15em;
        margin-top: 15px;
      }
    }
  }

  & i {
    font-size: 13rem;
    color: white;
    background-color: $primary-green;
    padding: 30px;
    border-radius: 50%;

    @include respond(xsphone) {
      padding: 20px;
    }
  }
}
