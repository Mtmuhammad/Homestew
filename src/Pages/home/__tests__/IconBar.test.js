import React from "react";
import { render, screen } from "@testing-library/react";
import IconBar from "../IconBar/IconBar";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<IconBar />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<IconBar />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render h3", () => {
  render(<IconBar />, { wrapper: BrowserRouter });
  const heading = screen.getByRole("heading", { name: /explore recipes/i });
  expect(heading).toBeInTheDocument();
});

describe("icon containers with icons", () => {
  test("should render icon containers", () => {
    render(<IconBar />, { wrapper: BrowserRouter });
    const iconContainers = screen.getAllByTestId("icon-container");
    expect(iconContainers.length).toBe(4);
    expect(iconContainers[0]).toHaveClass("icon");
    expect(iconContainers[1]).toHaveClass("icon");
    expect(iconContainers[2]).toHaveClass("icon");
    expect(iconContainers[3]).toHaveClass("icon");
  });

  test("should render icon descriptions", () => {
    render(<IconBar />, { wrapper: BrowserRouter });
    expect(screen.getByText(/bread recipes/i)).toBeInTheDocument();
    expect(screen.getByText(/main course/i)).toBeInTheDocument();
    expect(screen.getByText(/side dishes/i)).toBeInTheDocument();
    expect(screen.getByText(/desserts/i)).toBeInTheDocument();
  });

  test("should render icons", () => {
    render(<IconBar />, { wrapper: BrowserRouter });
    const icons = screen.getAllByTestId("icon");
    expect(icons.length).toBe(4);
    expect(icons[0]).toHaveClass("fa-bread-slice");
    expect(icons[1]).toHaveClass("fa-drumstick-bite");
    expect(icons[2]).toHaveClass("fa-bowl-food");
    expect(icons[3]).toHaveClass("fa-cookie");
  });
});
