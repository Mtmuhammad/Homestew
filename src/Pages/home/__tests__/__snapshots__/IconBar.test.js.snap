// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="icon-bar"
  >
    <div
      class="container"
      style="max-width: 1100px;"
    >
      <div
        class="row"
      >
        <h3
          class="tertiary text-center"
        >
          Explore Recipes:
        </h3>
        <div
          class="col-6 col-lg-3"
        >
          <div
            class="icon text-center"
            data-testid="icon-container"
          >
            <a
              href="/recipes/type/bread"
            >
              <i
                class="fa-solid fa-bread-slice"
                data-testid="icon"
              />
              <p>
                Bread Recipes
              </p>
            </a>
          </div>
        </div>
        <div
          class="col-6 col-lg-3"
        >
          <div
            class="icon text-center"
            data-testid="icon-container"
          >
            <a
              href="/recipes/type/main course"
            >
              <i
                class="fa-solid fa-drumstick-bite"
                data-testid="icon"
              />
              <p>
                Main Course
              </p>
            </a>
          </div>
        </div>
        <div
          class="col-6 col-lg-3"
        >
          <div
            class="icon text-center"
            data-testid="icon-container"
          >
            <a
              href="/recipes/type/side dish"
            >
              <i
                class="fa-solid fa-bowl-food"
                data-testid="icon"
              />
              <p>
                Side Dishes
              </p>
            </a>
          </div>
        </div>
        <div
          class="col-6 col-lg-3"
        >
          <div
            class="icon text-center"
            data-testid="icon-container"
          >
            <a
              href="/recipes/type/dessert"
            >
              <i
                class="fa-solid fa-cookie"
                data-testid="icon"
              />
              <p>
                Desserts
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
